dpnp/backend/kernels/dpnp_krnl_random.cpp

```diff
@@ -23,16 +23,51 @@
 // THE POSSIBILITY OF SUCH DAMAGE.
 //*****************************************************************************
 
+#include <mkl_vsl.h>
+#include <stdexcept>
+#include <vector>
+
 #include <dpnp_iface.hpp>
+
 #include "dpnp_fptr.hpp"
 #include "dpnp_utils.hpp"
 #include "queue_sycl.hpp"
 
-#include <vector>
-
 namespace mkl_rng = oneapi::mkl::rng;
 namespace mkl_blas = oneapi::mkl::blas;
 
+/**
+ * Use get/set functions to access/modify this variable
+ */
+static VSLStreamStatePtr rng_stream = nullptr;
+
+static void set_rng_stream(size_t seed = 1)
+{
+    if (rng_stream)
+    {
+        vslDeleteStream(&rng_stream);
+        rng_stream = nullptr;
+    }
+
+    vslNewStream(&rng_stream, VSL_BRNG_MT19937, seed);
+}
+
+static VSLStreamStatePtr get_rng_stream()
+{
+    if (!rng_stream)
+    {
+        set_rng_stream();
+    }
+
+    return rng_stream;
+}
+
+void dpnp_srand_c(size_t seed)
+{
+    backend_sycl::backend_sycl_rng_engine_init(seed);
+    set_rng_stream(seed);
+}
+
```

Comment:
I don't think that this is simpler.
I vote to save it as it was and revert you changes, except throw exception.